Sources/App/Commands/Analyze.swift

```diff
@@ -459,11 +459,20 @@ extension Analyze {
 
         let ageOfExistingVersion = Current.date().timeIntervalSinceReferenceDate - existingVersion.commitDate.timeIntervalSinceReferenceDate
 
-        // if existing version isn't older than our "window", keep it - otherwise
-        // use the latest incoming version
-        let resultingBranchVersion = ageOfExistingVersion < Constants.branchVersionRefreshDelay
-        ? existingVersion
-        : incomingVersion
+        let resultingBranchVersion: Version
+        if existingVersion.reference.branchName != incomingVersion.reference.branchName {
+            // if branch names differ we've got a renamed default branch and want to make
+            // sure it's updated as soon as possible -> no throttling
+            resultingBranchVersion = incomingVersion
+        } else {
+            // if existing version isn't older than our "window", keep it - otherwise
+            // use the latest incoming version
+            if ageOfExistingVersion < Constants.branchVersionRefreshDelay {
+                resultingBranchVersion = existingVersion
+            } else {
+                resultingBranchVersion = incomingVersion
+            }
+        }
```

Comment:
There's a bit of noise due to also fixing a typo. The essential change is in commit 710522880debfa70d5c8a06e69f37049d6b9dc76